scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/TaskInfoFactory.java

```diff
@@ -50,88 +56,182 @@ public TaskInfoFactory(ClusterState clusterState) {
         }
-        return ByteString.copyFromUtf8(writer.getBuffer().toString());
+        return builder
+                .build();
     }
 
-    private Protos.ExecutorInfo.Builder newExecutorInfo(Configuration configuration) {
-        Protos.ExecutorInfo.Builder executorInfoBuilder = Protos.ExecutorInfo.newBuilder()
-                .setExecutorId(Protos.ExecutorID.newBuilder().setValue(UUID.randomUUID().toString()))
-                .setFrameworkId(frameworkState.getFrameworkID())
-                .setName("elasticsearch-executor-" + UUID.randomUUID().toString())
-                .setCommand(newCommandInfo(configuration));
-        if (configuration.isFrameworkUseDocker()) {
-            Protos.ContainerInfo.DockerInfo.Builder containerBuilder = Protos.ContainerInfo.DockerInfo.newBuilder()
-                    .setImage(configuration.getExecutorImage())
-                    .setForcePullImage(configuration.getExecutorForcePullImage())
-                    .setNetwork(Protos.ContainerInfo.DockerInfo.Network.HOST);
-
-            executorInfoBuilder.setContainer(Protos.ContainerInfo.newBuilder()
-                    .setType(Protos.ContainerInfo.Type.DOCKER)
-                    .setDocker(containerBuilder)
-                    .addVolumes(Protos.Volume.newBuilder().setHostPath(CONTAINER_PATH_SETTINGS).setContainerPath(CONTAINER_PATH_SETTINGS).setMode(Protos.Volume.Mode.RO)) // Temporary fix until we get a data container.
-                    .addVolumes(Protos.Volume.newBuilder().setContainerPath(CONTAINER_DATA_VOLUME).setHostPath(configuration.getDataDir()).setMode(Protos.Volume.Mode.RW).build())
-                    .build())
-                    .addResources(Resources.cpus(configuration.getExecutorCpus(), configuration.getFrameworkRole()))
-                    .addResources(Resources.mem(configuration.getExecutorMem(), configuration.getFrameworkRole()))
-            ;
+    private Protos.CommandInfo nativeCommand(Configuration configuration, List<String> args) {
+        String address = configuration.getFrameworkFileServerAddress();
+        if (address == null) {
+            throw new NullPointerException("Webserver address is null");
         }
-        return executorInfoBuilder;
+        String httpPath = address + "/get/" + Configuration.ES_TAR;
+        String folders = configuration.getDataDir() + " " + HOST_SANDBOX;
+        String mkdir = "mkdir -p " + folders + "; ";
+        String chown = "chown -R nobody:nogroup " + folders + "; ";
+        String command = mkdir +
+                        chown +
+                " su -s /bin/sh -c \""
+                        + Configuration.ES_BINARY
+                        + " "
+                        + args.stream().collect(Collectors.joining(" "))
+                        + "\" nobody";
+        final Protos.Environment environment = Protos.Environment.newBuilder().addAllVariables(new ExecutorEnvironmentalVariables(configuration).getList()).build();
```

Comment:
Can the command generation code be moved to a method in the Configuration class? The command uses address, Configuration.BINARY, Configuration.ES_TAR and datadir so I think it should be moved there.